File: Team4.Seshat.Model/Models/Book.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Team4.Seshat.Model.Models
{
    [Table("BookRecord")]
    public class Book
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Author { get; set; }
        public int Quantity { get; set; }
        public int? BookGenreId { get; set; }
        [ForeignKey("BookGenreId")]
        public BookGenre Genre { get; set; }
        public int LibraryId { get; set; }
    }
}

File: front-end-ui/src/app/models/book.ts
import { BookGenre } from "./bookGenre";

export class Book {
    constructor(title: string, author: string, quantity: number) {
        this.title = title;
        this.author = author;
        this.quantity = quantity;
    }

    public title: string;
    public author: string;
    public quantity: number;
    public bookGenreId!: number;
}